### PR TITLE
New algorithm "Rename table field"

### PR DESCRIPTION
**Pull request by nyalldawson**
Takes an input layer, existing field and a new name for the field, and
outputs a new layer with the selected field renamed.

While this result could also be achieved with the Refactor Fields
algorithm, Refactor Fields isn't particularly model friendly. It
relies on a constant, fixed table structure, and can't adapt to
input tables with different field structures.

In constrast, this simple Rename Field algorithm adapts nicely for
model use, because it operates on a single field only and leaves
all the other fields untouched.

### CODE REVIEW

**Comment by nirvn** [comment]
Nice, I've often wished for exactly this.

File: src/analysis/processing/qgsalgorithmrenametablefield.cpp [hunk 1]
```diff
@@ -0,0 +1,115 @@
+/***************************************************************************
+                         qgsalgorithmrenametablefield.cpp
+                         -----------------------------------
+    begin                : November 2019
+    copyright            : (C) 2019 by Alexander Bruy
```
Comment (nirvn) [inline src/analysis/processing/qgsalgorithmrenametablefield.cpp]: Header copyright and email needs updating here?

File: src/analysis/processing/qgsalgorithmrenametablefield.h [hunk 2]
```diff
@@ -0,0 +1,66 @@
+/***************************************************************************
+                         qgsalgorithmrenametablefield.h
+                         ---------------------
+    begin                : December 2019
```
Comment (nirvn) [inline src/analysis/processing/qgsalgorithmrenametablefield.h]: January 2020?

File: src/analysis/processing/qgsalgorithmrenametablefield.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/analysis/processing/qgsalgorithmrenametablefield.cpp]: Thanks, fixed

**Review by alexbruy — APPROVED** [review]
Nice addition

**Comment by saberraz** [comment]
A bit out of context and in my wishlist, but would have been great to have a similar algorithm for In-place editing.

**Pull request merged**